File: IAIso-v5.0/core/iaiso-swift/Sources/IAIsoCore/StepOutcome.swift
import Foundation

/// Outcome of a single `PressureEngine.step()` call. The raw value is the wire form.
public enum StepOutcome: String, Sendable, CaseIterable {
    case ok = "ok"
    case escalated = "escalated"
    case released = "released"
    case locked = "locked"
}

File: IAIso-v5.0/core/iaiso-java/iaiso-core/src/main/java/io/iaiso/core/StepOutcome.java
package io.iaiso.core;

/**
 * The result of a single {@code step()} call. Wire-format strings,
 * lowercase, matching the other reference SDKs exactly.
 */
public enum StepOutcome {
    OK("ok"),
    ESCALATED("escalated"),
    RELEASED("released"),
    LOCKED("locked");

    private final String wireValue;

    StepOutcome(String wireValue) {
        this.wireValue = wireValue;
    }

    @Override
    public String toString() {
        return wireValue;
    }

    public static StepOutcome fromWire(String s) {
        for (StepOutcome o : values()) {
            if (o.wireValue.equals(s)) {
                return o;
            }
        }
        throw new IllegalArgumentException("unknown step outcome: " + s);
    }
}
